.changeset/wet-tables-pretend.md

```diff
@@ -0,0 +1,9 @@
+---
+"wrangler": patch
+---
+
+fix: abort some of the fetches when we unmount the `Remote` component.
+Adding abort to the `waitForPortToBeAvailable` as an option and in the `useEffect` the abort signal is called in the return which allows for graceful closing with no hanging
+of the `waitForPortToBeAvailable`
```

Comment:
```suggestion
fix: abort async operations in the `Remote` component to avoid unwanted side-effects

When the `Remote` component is unmounted, we now signal outstanding `fetch()` requests, and
`waitForPortToBeAvailable()` tasks to cancel them. This prevents unexpected requests from appearing
after the component has been unmounted, and also allows the process to exit cleanly without a delay.
```